gluten-core/src/main/scala/io/glutenproject/execution/BasicScanExecTransformer.scala

```diff
@@ -84,8 +84,8 @@ trait BasicScanExecTransformer extends TransformSupport {
       return false
```

Comment:
@zzcclp Looks this validation is also needed for CH backend. Is that correct?

Reply:
yes, CH backend needs this validation, the partition values are read from file path by CH backend.